modules/schema/src/main/java/org/apache/ignite/internal/schema/configuration/IndexValidatorImpl.java

```diff
@@ -64,28 +62,77 @@ public void validate(IndexValidator annotation, ValidationContext<NamedListView<
 
             if (tableView == null) {
                 ctx.addIssue(new ValidationIssue(key, "Unable to create index [name=" + key + "]. Table not found."));
+
+                // no further validation required for current index
+                continue;
             }
 
-            IndexDefinition index = SchemaConfigurationConverter.convert(idxView);
+            List<String> colocationColumns;
+            if (tableView.primaryKey() != null) {
```

Comment:
how it possible that PK is null?

Reply:
at this point we should not being bothered about presence or absence of primary key. All we need is to collect collocation columns, so let's work with what we've got

Reply:
My opion - it can give incorrect unerstanding possibilities of the systems during read the code. I don'tunderstand why we should care about such case here

Reply:
I'm not sure if it's correct to make assumptions about table restrictions when reading an index validator...